examples/adc_temperature.rs

```diff
@@ -0,0 +1,57 @@
+#![deny(unsafe_code)]
+#![no_main]
+#![no_std]
+
+use panic_semihosting as _;
+
+use stm32f1xx_hal::{
+    prelude::*,
+    pac,
+    adc,
+};
+use cortex_m_rt::{entry,exception,ExceptionFrame};
+
+use cortex_m_semihosting::hprintln;
+
+#[entry]
+fn main() -> ! {
+    // Aquire peripherals
+    let p = pac::Peripherals::take().unwrap();
+    let mut flash = p.FLASH.constrain();
+    let mut rcc = p.RCC.constrain();
+
+    // Several examples of HSE, PCLK1, ADC clocks
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(8.mhz()).pclk1(2.mhz()).adcclk(1.mhz()).freeze(&mut flash.acr);
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(16.mhz()).pclk1(4.mhz()).adcclk(2.mhz()).freeze(&mut flash.acr);
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(24.mhz()).pclk1(9.mhz()).adcclk(3.mhz()).freeze(&mut flash.acr);
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(24.mhz()).pclk1(12.mhz()).adcclk(4.mhz()).freeze(&mut flash.acr);
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(24.mhz()).pclk1(12.mhz()).adcclk(6.mhz()).freeze(&mut flash.acr);
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(36.mhz()).pclk1(12.mhz()).adcclk(6.mhz()).freeze(&mut flash.acr);
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(32.mhz()).pclk1(16.mhz()).adcclk(8.mhz()).freeze(&mut flash.acr);
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(40.mhz()).pclk1(20.mhz()).adcclk(10.mhz()).freeze(&mut flash.acr);
+    //let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(48.mhz()).pclk1(24.mhz()).adcclk(12.mhz()).freeze(&mut flash.acr);
```

Comment:
Are the comments here useful?

Reply:
Well, these are just several examples of clock settings that can be done with external oscillator. My main purpose was to demonstrate more or less the whole range of ADC frequencies. Besides, I think the example for use_hse is helpful regardless of adc usage.

Reply:
I removed those extra examples in comments. One use_hse example should be enough.

Reply:
I think it may be useful to document the relationships between the possible frequencies for the different registers in case there are some restrictions to the accepted values. This would clarify what you have to do if you want to set the ADC clock to 12 MHz, for example.

Going a bit further, I could also think it would be nice to document these value relationships in general here somewhere as a shortcut to some paragraph about prescaler config in the datasheet.

Reply:
I agree, but i'm not sure if this example is the right way to do this. I'd say it should either be in the documentation or a separate example.